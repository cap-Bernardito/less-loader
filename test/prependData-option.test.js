import {
  compile,
  getCodeFromBundle,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers';

jest.setTimeout(30000);

describe('prependData option', () => {
  it('should work prepend data as function', async () => {
    const testId = './prepend-data.less';
    const compiler = getCompiler(testId, {
      prependData() {
        return `@background: coral;`;
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work prepend data as string', async () => {
    const testId = './prepend-data.less';
    const compiler = getCompiler(testId, {
      prependData: `@background: coral;`,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
