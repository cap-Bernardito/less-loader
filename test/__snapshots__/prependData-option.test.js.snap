// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prependData option should work prepend data as function: css 1`] = `
".background {
  color: coral;
}
"
`;

exports[`prependData option should work prepend data as function: errors 1`] = `Array []`;

exports[`prependData option should work prepend data as function: warnings 1`] = `Array []`;

exports[`prependData option should work prepend data as string: css 1`] = `
".background {
  color: coral;
}
"
`;

exports[`prependData option should work prepend data as string: errors 1`] = `Array []`;

exports[`prependData option should work prepend data as string: warnings 1`] = `Array []`;
